browser/extensions/api/moonpay_api.cc

```diff
@@ -0,0 +1,90 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/extensions/api/moonpay_api.h"
+
+#include <string>
+#include <memory>
+#include <utility>
+
+#include "brave/browser/profiles/profile_util.h"
+#include "brave/components/moonpay/browser/regions.h"
+#include "brave/components/moonpay/common/pref_names.h"
+#include "brave/components/ntp_widget_utils/browser/ntp_widget_utils_region.h"
+#include "chrome/browser/profiles/profile.h"
+#include "components/prefs/pref_service.h"
+
+namespace {
+bool IsMoonpayAPIAvailable(content::BrowserContext* context) {
+  return brave::IsRegularProfile(context);
+}
+}
+
+namespace extensions {
+namespace api {
+
+ExtensionFunction::ResponseAction
+MoonpayIsBitcoinDotComSupportedFunction::Run() {
+  Profile* profile = Profile::FromBrowserContext(browser_context());
+
+  if (!IsMoonpayAPIAvailable(browser_context())) {
+    return RespondNow(Error("Not available in Tor/incognito/guest profile"));
+  }
+
+  bool is_supported = ntp_widget_utils::IsRegionSupported(
+      profile->GetPrefs(), moonpay::bitcoin_dot_com_supported_regions, true);
+  return RespondNow(OneArgument(
+      std::make_unique<base::Value>(is_supported)));
+}
+
+ExtensionFunction::ResponseAction
+MoonpayOnBuyBitcoinDotComCryptoFunction::Run() {
+  Profile* profile = Profile::FromBrowserContext(browser_context());
+
+  if (!IsMoonpayAPIAvailable(browser_context())) {
+    return RespondNow(Error("Not available in Tor/incognito/guest profile"));
+  }
+
+  profile->GetPrefs()->SetBoolean(kMoonpayHasBoughtBitcoinDotComCrypto, true);
+  profile->GetPrefs()->SetBoolean(kMoonpayHasInteractedBitcoinDotCom, true);
+
+  return RespondNow(NoArguments());
+}
+
+ExtensionFunction::ResponseAction
+MoonpayOnInteractionBitcoinDotComFunction::Run() {
+  Profile* profile = Profile::FromBrowserContext(browser_context());
+
+  if (!IsMoonpayAPIAvailable(browser_context())) {
+    return RespondNow(Error("Not available in Tor/incognito/guest profile"));
+  }
+
+  profile->GetPrefs()->SetBoolean(kMoonpayHasInteractedBitcoinDotCom, true);
+
+  return RespondNow(NoArguments());
+}
+
+ExtensionFunction::ResponseAction
+MoonpayGetBitcoinDotComInteractionsFunction::Run() {
```

Comment:
Doesn't seem like this, or the pref `kMoonpayHasInteractedBitcoinDotCom` is being actually used anywhere. Am I missing something?

Reply:
So we are capturing this now so that when the time comes to disable the widget after its run, it can be retained only for users who have interacted with it in some capacity.